### PR TITLE
adding clues & fault to onedrive restore

### PR DESCRIPTION
**Pull request by ryanfkeepers**
## Does this PR need a docs update or release note?

- [x] :no_entry: No 

## Type of change

- [x] :broom: Tech Debt/Cleanup

## Issue(s)

* #1970

## Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/internal/connector/onedrive/restore.go [hunk 1]
```diff
@@ -89,7 +89,7 @@ func getParentAndCollectionPermissions(
 
 	// TODO(ashmrtn): For versions after this pull the permissions from the
 	// current collection with Fetch().
-	colPerms, err = getParentPermissions(collectionPath, permissions)
+	colPerms, err := getParentPermissions(collectionPath, permissions)
```
Comment (meain) [inline src/internal/connector/onedrive/restore.go:92]: I guess we can removed the declaration of `colPerms` as well at top.

File: src/internal/connector/onedrive/restore.go [hunk 2]
```diff
@@ -210,20 +228,21 @@ func RestoreCollection(
...
+		ctx,
+		"destination_elements", restoreFolderElements,
+		"drive_id", drivePath.DriveID)
+
 	trace.Log(ctx, "gc:oneDrive:restoreCollection", directory.String())
-	logger.Ctx(ctx).Infow(
-		"restoring to destination",
-		"origin", dc.FullPath().Folder(false),
-		"destination", restoreFolderElements)
+	logger.Ctx(ctx).Info("restoring onedrive collection")
```
Comment (meain) [inline src/internal/connector/onedrive/restore.go:237]: Should we be reducing the info in log messages?

File: src/internal/connector/onedrive/restore.go [hunk 3]
```diff
@@ -431,42 +450,31 @@ func CreateRestoreFolders(
...
-			support.ConnectorStackErrorTrace(err),
-		)
+		return "", clues.Wrap(err, "getting drive root").WithClues(ctx).With(graph.ErrData(err)...)
 	}
 
-	logger.Ctx(ctx).Debugf("Found Root for Drive %s with ID %s", driveID, *driveRoot.GetId())
+	parentFolderID := ptr.Val(driveRoot.GetId())
+	ctx = clues.Add(ctx, "drive_root_id", parentFolderID)
+
+	logger.Ctx(ctx).Debug("found drive root")
```
Comment (meain) [inline src/internal/connector/onedrive/restore.go:459]: Same question about reducing the information that we print out in logs.

**Review by meain — APPROVED** [review]
Few small nits.

File: src/internal/connector/onedrive/restore.go (hunk 2 above)
Comment (ryanfkeepers) [reply src/internal/connector/onedrive/restore.go:237]: See line 228.  This data was already added to the context.

**Review by meain — APPROVED** [review]

**Pull request merged**